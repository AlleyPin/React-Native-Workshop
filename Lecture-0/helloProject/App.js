import React from 'react';
import { FlatList, StyleSheet, Text, View, TouchableOpacity } from 'react-native';

export default class App extends React.Component {
  render() {
    return (
      <View style={styles.container} serviceType={'food'}>
        <View style={styles.header}></View>
        <Text>Open up App.js to start working on your app!</Text>
        <Text>Changes you make will automatically reload.</Text>
        <Text>Shake your phone to open the developer menu.</Text>
        <FlatList
          data={[{key: 'a'}, {key: 'b'}]}
          renderItem={function(element) {
            return <TouchableOpacity style={styles.cell} onPress={() => {
              console.log(element)
            }}><Text>{element.item.key}</Text></TouchableOpacity> 
          }}
        >
        </FlatList>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  header: {
    marginTop: 64
  },
  cell: {
    height: 100,
    borderWidth: 1,
    justifyContent: 'center',
    alignItems: 'center'
  }
});
