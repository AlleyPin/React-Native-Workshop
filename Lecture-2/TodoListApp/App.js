import React from 'react';
import { StyleSheet, View, FlatList } from 'react-native';
import Expo from 'expo';
import ListItem from './components/ListItem';

export default class App extends React.Component {
  renderItem = (element) => (
    <ListItem
      item={element.item}
    />
  );

  render() {
    return (
      <View style={styles.container}>
        <FlatList
          data={[{ key: 'item 1' }, { key: 'item 2' }]}
          renderItem={(element) => this.renderItem(element)}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: Expo.Constants.statusBarHeight
  }
});
