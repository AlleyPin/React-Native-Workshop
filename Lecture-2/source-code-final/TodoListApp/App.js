import React from 'react';
import { StyleSheet, Button, FlatList, TextInput, View } from 'react-native';
import Expo from 'expo';
import ListItem from './components/ListItem';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      items: [{ title: 'item 1' }, { title: 'item 2' }],
      addItemTextInputValue: ''
    };
  }

  onAddItemTextChanged = value => {
    this.setState({
      addItemTextInputValue: value
    });
  };

  addItem = () => {
    if (!this.state.addItemTextInputValue) return;

    this.setState({
      items: [...this.state.items, { title: this.state.addItemTextInputValue }],
      addItemTextInputValue: ''
    });
  };

  deleteItem = title => {
    this.setState({
      items: this.state.items.filter(item => item.title !== title)
    });
  };

  renderItem = (element) => (
    <ListItem
      item={element.item}
      deleteItem={this.deleteItem}
    />
  );

  render() {
    return (
      <View style={styles.container}>
        <FlatList
          data={this.state.items}
          renderItem={element => this.renderItem(element)}
          keyExtractor={item => item.title}
        />
        <View style={styles.addItemContainer}>
          <TextInput
            style={styles.textInput}
            placeholder="Type item name here"
            value={this.state.addItemTextInputValue}
            onChangeText={this.onAddItemTextChanged}
          />
          <Button
            title="Add Item"
            color="#FDDA3B"
            onPress={this.addItem}
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: Expo.Constants.statusBarHeight
  },
  addItemContainer: {
    flexDirection: 'row',
    margin: 24
  },
  textInput: {
    flex: 1,
    marginRight: 12,
    fontSize: 16
  }
});
