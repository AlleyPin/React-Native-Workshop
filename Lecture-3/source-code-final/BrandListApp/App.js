import React from 'react';
import { StyleSheet, Button, FlatList, TextInput, View } from 'react-native';
import Expo from 'expo';
import ListItem from './components/ListItem';
import { mockBrands } from './seeds';

export default class App extends React.Component {
  state = {
    items: mockBrands,
    addItemTextInputValue: ''
  }

  onAddItemTextChanged = value => {
    this.setState({
      addItemTextInputValue: value
    });
  };

  addItem = () => {
    if (!this.state.addItemTextInputValue) return;

    this.setState({
      items: [...this.state.items, { name: this.state.addItemTextInputValue }],
      addItemTextInputValue: ''
    });
  };

  deleteItem = title => {
    this.setState({
      items: this.state.items.filter(item => item.title !== title)
    });
  };

  renderItem = (element) => (
    <ListItem
      item={element.item}
      deleteItem={this.deleteItem}
    />
  );

  render() {
    return (
      <View style={styles.container}>
        <FlatList
          data={this.state.items}
          renderItem={element => this.renderItem(element)}
          keyExtractor={item => item.name}
        />
        <View style={styles.addItemContainer}>
          <TextInput
            style={styles.textInput}
            placeholder="Type item name here"
            value={this.state.addItemTextInputValue}
            onChangeText={this.onAddItemTextChanged}
          />
          <Button
            title="Add Item"
            color="#FDDA3B"
            onPress={this.addItem}
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: Expo.Constants.statusBarHeight
  },
  addItemContainer: {
    flexDirection: 'row',
    margin: 24
  },
  textInput: {
    flex: 1,
    marginRight: 12,
    fontSize: 16
  }
});
