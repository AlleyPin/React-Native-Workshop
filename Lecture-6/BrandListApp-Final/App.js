import React from 'react';
import { Provider } from 'react-redux';
import BrandScreen from './BrandScreen';
import store from './store';

class App extends React.Component {
  render() {
    return (
      <Provider store={store}>
        <BrandScreen />
      </Provider>
    );
  }
}

export default App;

