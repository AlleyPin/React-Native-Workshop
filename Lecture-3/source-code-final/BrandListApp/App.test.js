import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

it('renders correctly', () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it('onAddItemTextChanged change state addItemTextInputValue', () => {
  const rendered = renderer.create(<App />);
  const instance = rendered.getInstance();
  instance.setState = jest.fn();

  instance.onAddItemTextChanged(3);

  expect(instance.setState).toBeCalledWith({
    addItemTextInputValue: 3
  });
});
