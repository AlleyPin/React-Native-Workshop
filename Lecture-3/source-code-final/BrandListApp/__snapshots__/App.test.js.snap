// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginTop": 20,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "title": "item 1",
        },
        Object {
          "title": "item 2",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "yellow",
                "height": 160,
                "justifyContent": "center",
                "padding": 20,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "borderWidth": 1,
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              圖片
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderWidth": 1,
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "borderWidth": 1,
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              xxx義大利麵店
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "borderWidth": 1,
                }
              }
            >
              6%的反現
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "yellow",
                "height": 160,
                "justifyContent": "center",
                "padding": 20,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "borderWidth": 1,
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              圖片
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderWidth": 1,
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "borderWidth": 1,
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              xxx義大利麵店
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "borderWidth": 1,
                }
              }
            >
              6%的反現
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "flexDirection": "row",
        "margin": 24,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Type item name here"
      style={
        Object {
          "flex": 1,
          "fontSize": 16,
          "marginRight": 12,
        }
      }
      value=""
    />
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#FDDA3B",
              },
            ]
          }
        >
          Add Item
        </Text>
      </View>
    </View>
  </View>
</View>
`;
